app/assets/javascripts/discourse/app/components/glimmer-topic-timeline.js

```diff
@@ -0,0 +1,94 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import optionalService from "discourse/lib/optional-service";
+import { inject as service } from "@ember/service";
+
+export default class GlimmerTopicTimeline extends Component {
+  @service site;
+  @service siteSettings;
+  @service currentUser;
+
+  @tracked dockAt = null;
+  @tracked dockBottom = null;
+  @tracked enteredIndex = this.args.enteredIndex;
+
+  adminTools = optionalService();
+  intersectionObserver = null;
+
+  constructor() {
+    super(...arguments);
+
+    if (this.args.prevEvent) {
+      this.enteredIndex = this.args.prevEvent.postIndex - 1;
+    }
+
+    if (!this.site.mobileView) {
+      this.intersectionObserver = new IntersectionObserver((entries) => {
+        for (const entry of entries) {
+          const bounds = entry.boundingClientRect;
+
+          if (entry.target.id === "topic-bottom") {
+            this.topicBottom = bounds.y + window.scrollY;
+          } else {
+            this.topicTop = bounds.y + window.scrollY;
+          }
+        }
+      });
+
+      const elements = [
+        document.querySelector(".container.posts"),
+        document.querySelector("#topic-bottom"),
+      ];
+
+      for (let i = 0; i < elements.length; i++) {
+        this.intersectionObserver.observe(elements[i]);
+      }
+    }
+  }
+
+  get displaySummary() {
+    this.siteSettings.summary_timeline_button &&
```

Comment:
```suggestion
    return this.siteSettings.summary_timeline_button &&
```

(The perils of jumping between Ruby and JS development - I do this all the time 😅)